### PR TITLE
Implement mania normal-note hit-explosion skinning

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu-framework/pull/3418
- [x] https://github.com/ppy/osu/pull/8555
- [x] https://github.com/ppy/osu-resources/pull/95

### CODE REVIEW

File: osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            if (explosion != null)
+                explosion.Anchor = direction.NewValue == ScrollingDirection.Up ? Anchor.TopCentre : Anchor.BottomCentre;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            explosion?.FadeInFromZero(80)
+                     .Then().FadeOut(120);
```
Comment (mcendu) [inline osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs:66]:
```suggestion
            explosion?.FadeInFromZero(80)
                     .Then().FadeOut(120);

            Expire(true);
```

File: osu.Game.Rulesets.Mania/UI/Components/ColumnHitObjectArea.cs [hunk 2]
```diff
@@ -12,7 +11,7 @@ namespace osu.Game.Rulesets.Mania.UI.Components
 {
     public class ColumnHitObjectArea : HitObjectArea
     {
-        public readonly Container<HitExplosion> Explosions;
+        public readonly Container<Drawable> Explosions;
```
Comment (mcendu) [inline osu.Game.Rulesets.Mania/UI/Components/ColumnHitObjectArea.cs:14]:
```suggestion
        public readonly Container Explosions;
```

Same for other occurences.

**Review by mcendu — CHANGES_REQUESTED** [review]
A few nits

File: osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs:66]: Good catch.

File: osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+                d.Origin = Anchor.Centre;
+                d.Blending = BlendingParameters.Additive;
+                d.Scale = new Vector2(explosionScale);
+
+                if (!(d is TextureAnimation texAnimation))
+                    return;
+
+                if (texAnimation.FrameCount > 0)
+                    texAnimation.DefaultFrameLength = Math.Max(texAnimation.DefaultFrameLength, 170.0 / texAnimation.FrameCount);
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs:48]: isn't this too late? `DefaultFrameLength` only applies to frames which are added from this point forward.

Comment (smoogipoo) [reply osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs:48]: Ah, didn't realise. Would you be against adding another optional parameter to `GetAnimation()` for the frame length, alongside the current `applyConfigFrameRate`?

Comment (peppy) [reply osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs:48]: a similar thing seems to be required for sliderball too. i mean we already have 8 odd parameters so why not?

Comment (smoogipoo) [reply osu.Game.Rulesets.Mania/Skinning/LegacyHitExplosion.cs]: Made the change, I don't think it's _too_ bad...

**Review by peppy — APPROVED** [review]

**Pull request merged**